cEP-0028.md

```diff
@@ -0,0 +1,150 @@
+# Optimizing coala for language server
+
+| Metadata |                                       |
+| -------- | ------------------------------------- |
+| cEP      | 0028                                  |
+| Version  | 0.1                                   |
+| Title    | Optimizing coala for language server  |
+| Authors  | Kilari Teja <mailto:ksdme9@gmail.com> |
+| Status   | Proposed                              |
+| Type     | Process                               |
+
+## Abstract
+
+This cEP describes the optimizations that need to be carried out
+to improve the performance of the coala based language server.
+
+## Introduction
+
+Language servers are the tools used to serve the code editors with
+functionalities such as linting, code fixing, analysis etc in a
+standalone and independent manner. coala as a code style checker,
+linting tool and code fixer could be a powerful tool to be integrated
+into the code editor.
+
+Code Editing is a continuous process and the editors are expected
+to provide feedback on the fly, coala on the other hand was primarily
+designed as an analysis tool to be used after an incremental update to
+the codebase. Thus, the current implementation of a coala language
+server is severely limited by its performance.
+
+The performance of the server can be improved by primarily optimizing
+the coala and language server interface and improving the bear configuration
+options to avoid long running bears.
+
+## Proposal
+
+The optimization process focuses to maintain minimal changes to coala core
+and rather optimize the coala and language server interface. The following
+are the proposed ideas.
+
+1. Extending coala configuration parameters
+   Some bears are not often required to be executed on every cycle of coala
+   analysis especially during use cases such as this. The parameter set can
+   thus be extended to support parameters of type `cls_*`. Since these
```

Comment:
No `cls_` . New settings need to describe what they do , and be reusable.

Reply:
`cls_` was proposed to keep those parameters clearly marked as coala-ls configuration parameters.

Reply:
coala-ls is not different from another editor. The needs are the same. All editor plugins should be able to use these settings to improve the usability and interactivity for their users.